rllib/utils/replay_buffers/replay_buffer.py

```diff
@@ -0,0 +1,206 @@
+
+        Returns:
+            Concatenated batch of items. None if buffer is empty.
+        """
+        # If we don't have any samples yet in this buffer, return None.
+        if len(self) == 0:
+            return None
+
+        idxes = [random.randint(0, len(self) - 1) for _ in range(num_items)]
+        sample = self._encode_sample(idxes)
+        # Update our timesteps counters.
+        self._num_timesteps_sampled += len(sample)
+        return sample
+
+    @ExperimentalAPI
+    @DeveloperAPI
+    def stats(self, debug: bool = False) -> dict:
+        """Returns the stats of this buffer.
+
+        Args:
+            debug: If True, adds sample eviction statistics to the returned
+                stats dict.
+
+        Returns:
+            A dictionary of stats about this buffer.
+        """
+        data = {
+            "added_count": self._num_timesteps_added,
+            "added_count_wrapped": self._num_timesteps_added_wrap,
+            "eviction_started": self._eviction_started,
+            "sampled_count": self._num_timesteps_sampled,
+            "est_size_bytes": self._est_size_bytes,
+            "num_entries": len(self._storage),
+        }
+        if debug:
+            data.update(self._evicted_hit_stats.stats())
+        return data
+
+    @ExperimentalAPI
+    @DeveloperAPI
+    def get_state(self) -> Dict[str, Any]:
+        """Returns all local state.
+
+        Returns:
+            The serializable local state.
+        """
+        state = {"_storage": self._storage, "_next_idx": self._next_idx}
+        state.update(self.stats(debug=False))
+        return state
+
+    @ExperimentalAPI
+    @DeveloperAPI
+    def set_state(self, state: Dict[str, Any]) -> None:
+        """Restores all local state to the provided `state`.
+
+        Args:
+            state: The new state to set this buffer. Can be
+                obtained by calling `self.get_state()`.
+        """
+        # The actual storage.
+        self._storage = state["_storage"]
+        self._next_idx = state["_next_idx"]
+        # Stats and counts.
+        self._num_timesteps_added = state["added_count"]
+        self._num_timesteps_added_wrap = state["added_count_wrapped"]
+        self._eviction_started = state["eviction_started"]
+        self._num_timesteps_sampled = state["sampled_count"]
+        self._est_size_bytes = state["est_size_bytes"]
+
+    def _encode_sample(self, idxes: List[int]) -> SampleBatchType:
+        out = SampleBatch.concat_samples([self._storage[i] for i in idxes])
+        out.decompress_if_needed()
+        return out
+
+    def get_host(self) -> str:
+        """Returns the computer's network name.
+
+        Returns:
+            The computer's networks name or an empty string, if the network
+            name could not be determined.
+        """
+        return platform.node()
+
+# TODO: What about this? Ask Sven?:
+# Visible for testing.
+# _local_replay_buffer = None
```

Comment:
Let's get rid of it and see what happens. :)
This was solely for testing purposes anyways and is not used in any of our algorithms.